src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,47 @@
-import React from 'react';
-import { Loader } from '../Loader';
+import React, { useState, useEffect } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { Loader } from '../Loader/Loader';
+
+type Props = {
+  todos: Todo[];
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  setOnSelect: React.Dispatch<React.SetStateAction<number>>;
+  onSelect: number;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todos,
+  setOpen,
+  setOnSelect,
+  onSelect,
+}) => {
+  const select = todos.find((todo) => todo.id === onSelect);
+  const [user, setUser] = useState<User>({
+    id: 0,
+    name: '',
+    email: '',
+    phone: '',
+  });
+  const [isUserInfoLoaded, setIsUserInfoLoaded] = useState(false);
+
+  useEffect(() => {
+    getUser(select?.userId)
+      .then((userFromServer) => {
+        setUser(userFromServer);
+        setIsUserInfoLoaded(true);
+      })
+      .catch((error) => {
+        throw new Error(error);
+      });
```

Comment:
Is there any sense in throwing error here in case if error is thrown before? 🙂 
It's absolutely not informative for users, we should show some kind of message with error text to explain user what was wrong (at least some red text instead of user info

Reply:
\+